web/settings.gradle.kts

```diff
@@ -1,22 +1,50 @@
+
 pluginManagement {
+    val COMPOSE_CORE_VERSION: String by settings
+    println("[build] compose core version: $COMPOSE_CORE_VERSION")
+
+    // pluginManagement section won't see outer scope, hence the FQ names
+    fun properties(path: String): java.util.Properties? {
+        val localPropertiesFile = File(path)
+        if (!localPropertiesFile.exists()) {
+            return null
+        }
+        return java.io.FileInputStream(localPropertiesFile).use() { inputStream ->
+            val props = java.util.Properties()
+            props.load(inputStream)
+            props
+        }
+    }
+
+    val localProperties: java.util.Properties? = properties("local.properties")
+    
+
+    val repos = (localProperties?.getProperty("compose.web.repos"))?.split(File.pathSeparator)
+
     repositories {
         gradlePluginPortal()
         mavenCentral()
-        mavenLocal()
+
+        repos?.forEach { urlPath ->
+            maven {
+                url = uri(urlPath)
+            }
+        }
+
         maven {
             url = uri("https://maven.pkg.jetbrains.space/public/p/compose/dev")
         }
         maven {
             url = uri("https://packages.jetbrains.team/maven/p/ui/dev")
         }
+
         google()
     }
 
     resolutionStrategy {
         eachPlugin {
             if (requested.id.id == "org.jetbrains.compose") {
-                println("[build] compose core version: ${extra["COMPOSE_CORE_VERSION"]}")
```

Comment:
is it okay to keep this println? sometimes it helps  to understand that I'm using correct version.

Reply:
This message just moved upper (and printed once) since we have common plugin resolution for all subprojects.